File: C/Colleen/Colleen.c
#include <stdio.h>

void f1()
{
	const char *val = "#include <stdio.h>%3$c%3$cvoid f1()%3$c{%3$c%4$cconst char *val = %1$c%2$s%1$c;%3$c%4$cprintf(val, 34, val, 10, 9);%3$c}%3$c%3$c/*%3$c  This program will display itself when run%3$c*/%3$c%3$cint main()%3$c{%3$c/*%3$c  Function to display itself%3$c*/%3$c%4$cf1();%3$c}%3$c";
	printf(val, 34, val, 10, 9);
}

/*
  This program will display itself when run
*/

int main()
{
/*
  Function to display itself
*/
	f1();
}

File: Python/Colleen/Colleen.py
#!/usr/bin/python3
VALUE="#!/usr/bin/python3{nl}VALUE={quo}{val}{quo}{nl}# This program will display itself when run{nl}def hello():{nl}    print(VALUE.format(nl=chr(10),quo=chr(34),val=VALUE)){nl}if __name__ == {quo}__main__{quo}:{nl}    # Function to display itself{nl}    hello()"
# This program will display itself when run
def hello():
    print(VALUE.format(nl=chr(10),quo=chr(34),val=VALUE))
if __name__ == "__main__":
    # Function to display itself
    hello()
